services/motion/builtin/move_request.go

```diff
@@ -262,54 +267,41 @@ func (mr *moveRequest) obstaclesIntersectPlan(
 				return state.ExecuteResponse{}, err
 			}
 
-			// build representation of frame system's inputs
-			currentInputs, err := mr.kinematicBase.CurrentInputs(ctx)
-			if err != nil {
-				return state.ExecuteResponse{}, err
-			}
-			inputMap := referenceframe.StartPositions(mr.planRequest.FrameSystem)
-			inputMap[mr.kinematicBase.Name().ShortName()] = currentInputs
-
-			// get the current position of the base
-			currentPosition, err := mr.kinematicBase.CurrentPosition(ctx)
+			// get the execution state of the base
+			baseExecutionState, err := mr.kinematicBase.ExecutionState(ctx)
 			if err != nil {
 				return state.ExecuteResponse{}, err
 			}
 
-			// Note: the value of wayPointIndex is subject to change between when this function is first entered
-			// versus when CheckPlan is actually called.
-			// We load the wayPointIndex value to ensure that all information is up to date.
-
-			// get the pose difference between where the robot is versus where it ought to be.
-			errorState, err := mr.kinematicBase.ErrorState(ctx)
-			if err != nil {
-				return state.ExecuteResponse{}, err
-			}
+			// build representation of frame system's inputs
+			// TODO(pl): in the case where we have e.g. an arm (not moving) mounted on a base, we should be passing its current
+			// configuration rather than the zero inputs
+			inputMap := referenceframe.StartPositions(mr.planRequest.FrameSystem)
```

Comment:
I am not too crazy about having this be here since we already do `mr.kinematicBase.ExecutionState`, but it makes sense for us to do since we need to the starting inputs of the other frames..
Maybe we can add an `extras` param to `mr.kinematicBase.ExecutionState` so that we do not need to another call/augmentation here.

Reply:
Once we are moving more than one thing at a time, we will need to fuse the execution states of all relevant pieces.